syntax = "proto3";
package MarketProto;

message OrderMSG {
  string symbol = 1;
  bool buySide = 2; 
  int64 quantity = 3;
  int64 price = 4;
}

message OrderFillMSG {
  int64 orderID = 1;
  int64 filled = 2;
}

message OrderResponseMSG {
  bool successful = 1;
  optional int64 orderID = 2;
  int64 amountFilled = 3;
  int64 price = 4;
}

message SignupMSG{
  string name = 1;
}

message SignupResponseMSG {
  int64 assignedId = 1;
}

message MSGConfirmMSG {
  int64 messageID = 1;
}